doc/optimizers.md

```diff
@@ -1394,6 +1394,114 @@ double Optimize(arma::mat& X);
  * [Semidefinite programs](#semidefinite-programs) (includes example usage of `PrimalDualSolver`)
 
+## Quasi-Hyperbolic Momentum Update
+
+ *An optimizer for [differentiable separable functions](#differentiable-separable-functions).*
+
+ Quasi Hyperbolic Momentum Update is an update policy for SGD where the Quasi Hyperbolic terms are added to the
+ parametrisation. Simply put QHM’s update rule is a weighted average of momentum’s and plain SGD’s
+ update rule.
+
+#### Constructors
+
+  * `QHSGD()`
+  * `QHSGD(`_`stepSize, batchSize`_`)`
+  * `QHSGD(`_`stepSize, batchSize, maxIterations, tolerance, shuffle`_`)`
+
+ Note that `QHSGD` is based on the templated type
+ `SGD<`_`UpdatePolicyType, DecayPolicyType`_`>` with _`UpdatePolicyType`_` =
+ QHUpdate` and _`DecayPolicyType`_` = NoDecay`.
+
+#### Attributes
+
+ | **type** | **name** | **description** | **default** |
+ |----------|----------|-----------------|-------------|
+ | `double` | **`stepSize`** | Step size for each iteration. | `0.01` |
+ | `size_t` | **`batchSize`** | Batch size to use for each step. | `32` |
+ | `size_t` | **`maxIterations`** | Maximum number of iterations allowed (0 means no limit). | `100000` |
+ | `double` | **`tolerance`** | Maximum absolute tolerance to terminate algorithm. | `1e-5` |
+ | `bool` | **`shuffle`** | If true, the function order is shuffled; otherwise, each function is visited in linear order. | `true` |
+
+ Attributes of the optimizer may also be modified via the member methods
+ `StepSize()`, `BatchSize()`, `MaxIterations()`, `Tolerance()`, and `Shuffle()`.
+
+ Note that the `QHUpdate` class has the constructor  `QHUpdate(`_`v, momentum`_`)` with a
+ default value of `0.7` for the quasi hyperbolic term and `0.999` for the momentum term.
+
+#### Examples
+
+ ```c++
+ RosenbrockFunction f;
+ arma::mat coordinates = f.GetInitialPoint();
+
+ QHSGD optimizer(0.01, 32, 100000, 1e-5, true);
+ optimizer.Optimize(f, coordinates);
+ ```
+
+#### See also:
+
+  * [Quasi-Hyperbolic Momentum and Adam For Deep Learning](https://arxiv.org/pdf/1810.06801.pdf)
+  * [Momentum SGD](#momentum-sgd)
+  * [Nesterov Momentum SGD](#nesterov-momentum-sgd)
+  * [SGD in Wikipedia](https://en.wikipedia.org/wiki/Stochastic_gradient_descent)
+  * [Differentiable separable functions](#differentiable-separable-functions)
+
+## QHAdam
+
+ *An optimizer for [differentiable separable functions](#differentiable-separable-functions).*
+
+ QHAdam is an optimizer which implements the QHAdam Adam algorithm
+ which uses Quasi-Hyperbolic Descent with the Adam Optimizer. This method is the Adam variant of the quasi-hyperbolic update for Adam.
+ It replaces the moment estimators of Adam with Quasi - Hyperbolic terms , and with
+ different values of those terms can recover the following Adam Polices
+ QHAdam recovers Adam when ν1 = ν2 = 1
+ QHAdam recovers RMSProp when ν1 = 0 and ν2 = 1
+ QHAdam reovers NAdam when ν1 = β1 and ν2 = 1
```

Comment:
I think we can do a little cleanup here too:

```
QHAdam is an optimizer that uses quasi-hyperbolic descent with the Adam optimizer.  This replaces the moment estimators of Adam with quasi-hyperbolic terms, and different values of the `v1` and `v2` parameters are equivalent to the following other optimizers:

 * When `v1 = v2 = 1`, `QHAdam` is equivalent to `Adam`.
 * When `v1 = 0` and `v2 = 1`, `QHAdam` is equivalent to `RMSProp`.
 * When `v1 = beta1` and `v2 = 1`, `QHAdam` is equivalent to `Nadam`.
```

I really like the reductions in the documentation.